### PR TITLE
feat(TDC-3961): Text mode for file input fields

### PR DESCRIPTION
**Pull request by olliew-talend**
**What is the problem this PR is trying to solve?**
The file input component does not currently register a text-only display mode which results in missing information when viewing pre-filled form fields in read-only contexts.

**What is the chosen solution to this problem?**
Register a new default text mode function for file input fields.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

**Notes:**
Any forms currently used with a display mode of text, that include a file input, will now automatically show the contents of the file input in text mode. This results in more content being rendered to the screen than may have been intended or designed for. In addition, it will result in the display of the file name value of the input, which may not be desirable in some cases.

### CODE REVIEW

File: packages/forms/src/UIForm/utils/widgets.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const widgets = {
 	code_text: CodeTextMode,
 	comparator_text: TextModeComparator,
 	datalist_text: DatalistTextMode,
-	file_text: () => null,
+	file_text: FileTextMode,
```
Comment (frassinier) [inline packages/forms/src/UIForm/utils/widgets.js:59]: Not a breaking change imho, you simply show what was previously hidden

**Review by frassinier — APPROVED** [review]

File: packages/forms/src/UIForm/utils/widgets.js (hunk 1 above)
Comment (olliew-talend) [reply packages/forms/src/UIForm/utils/widgets.js:59]: Thanks buddy. I'm happy to not call it a BC if that's the consensus. I suppose I look at this kind of thing as if it's a new feature which forces an unrequested change in behaviour then it is breaking to the experience, even if it doesn't cause errors.

Comment (frassinier) [reply packages/forms/src/UIForm/utils/widgets.js:59]: Actually, imagine CSS changes, they could be UI/UX huge changes but the apps will continue to work without any developer manipulations 🤷

Comment (olliew-talend) [reply packages/forms/src/UIForm/utils/widgets.js:59]: Agree, which is why I consider CSS changes to be breaking too! :grin:

I suppose it's just the definition of 'breaking' as 'causes errors' or 'impacts the experience'. I suppose my view is a result of my background :shrug:

**Review by ncomont — APPROVED** [review]

**Pull request merged**